lib/Parse/ParsePattern.cpp

```diff
@@ -888,6 +879,56 @@ void Parser::parseAsyncThrows(
   }
 }
 
+/// Parse a type in parentheses after a `throws` or `rethrows`
+///
+ParserResult<TypeRepr> Parser::parseThrowsType() {
+  // look for the left paren for throws (_type_)
+  BacktrackingScope backtrack(*this);
+  
+  ParserResult<TypeRepr> throwsTypeRepr;
+  
+  if (Tok.is(tok::l_paren)) {
+    consumeToken();
+    throwsTypeRepr = parseTypeIdentifier();
```

Comment:
```suggestion
    throwsTypeRepr = parseTypeSimpleOrComposition();
```

I would suggest using `parseTypeSimpleOrComposition()` as it will parse types like `[Bar]` as well (which theoretically can conform to `Error`)

Reply:
I can change it and test it. I need to improve the tests in general for this.